modules/lang-painless/src/test/java/org/elasticsearch/painless/AnalysisScriptTests.java

```diff
@@ -0,0 +1,32 @@
+package org.elasticsearch.painless;
+
+import org.elasticsearch.painless.spi.Whitelist;
+import org.elasticsearch.script.AnalysisPredicateScript;
+import org.elasticsearch.script.ScriptContext;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AnalysisScriptTests extends ScriptTestCase {
```

Comment:
I think this should be in your plugin instead of painless if we can manage it.

Reply:
I don't follow - do you mean moving the analysis script context out of core and into the common analysis plugin, or do you mean creating an entirely new plugin just for this token filter (and any other script-based filters we come up with)?

Reply:
I mean I feel like it should live in the common analysis plugin and not in the server at all. Since it is a new thing I figure we should isolate it as much as we can so we know when stuff is using it. Not because it is dirty or anything, just because a smaller core is easier to reason about. And keeping this entirely within the analysis-common plugin helps to prove out the work that we've done for plugins extending painless.

Reply:
The tricky part here will be unit testing, as ScriptTestCase is a painless-specific class.  The painless extension module uses rest tests only, but I'd much rather have compiled unit tests here.  Maybe we should build a separate painless-testing module, that can be used for tests elsewhere?

Reply:
In general we try to unit test this sort of thing without painless and use a mock script engine instead that just calls code on the test class. It is totally reasonable to have a few integration tests that do use painless though.